metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -716,6 +722,47 @@ class MetalsLanguageServer(
     onChange(paths).asJava
   }
 
+  private def insertPackage(path: AbsolutePath): Unit = {
+    def createPackageName(path: AbsolutePath): Option[String] = {
+      if (path.toFile.length() == 0) {
+        val sourceDirectory = buildTargets.inverseSourceDirectory(path)
+        val relativeDirectory = sourceDirectory
+          .map(path.toRelative)
+          .flatMap(relativePath => Option(relativePath.toFile.getParentFile))
+          .map(_.getPath)
+        val packageName = relativeDirectory.map(_.replace("/", "."))
+
+        packageName.map("package " + _)
+      } else {
+        None
+      }
+    }
+
+    def createPackageWorkspaceEdit(packageName: String): WorkspaceEdit = {
+      val textEdit = new TextEdit(
+        new Range(new Position(0, 0), new Position(0, 0)),
+        s"$packageName\n"
+      )
+
```

Comment:
I would loose the newlines - they don't add much to readability

Reply:
Are these the newlines after the package name? My expectation for this would be to have the cursor in this position after the package insertion:

```scala
package com.foo.bar

<CURSOR>
```

Reply:
what about:
```scala
val textEdits = Collections.singletonList(textEdit)
val changes = Collections.singletonMap(path.toString, textEdits)
new WorkspaceEdit(changes)
```

or going through Scala collections:
```scala
val textEdits = List(textEdit).asJava
val changes = Map(path.toString, textEdits).asJava
new WorkspaceEdit(changes)
```